cmd/wit-bindgen-go/cmd/generate/generate.go

```diff
@@ -159,5 +182,32 @@ func action(ctx context.Context, cmd *cli.Command) error {
 		}
 	}
 
+	witDir := filepath.Join(out, "wit")
+	err = os.MkdirAll(witDir, outPerm)
+	if err != nil {
+		return err
+	}
+	witFilePath := filepath.Join(witDir, "webassembly.wit.wasm")
+
+	fmt.Fprintf(os.Stderr, "Generated WIT file: %s\n", witFilePath)
+
+	wasmTools, err := exec.LookPath("wasm-tools")
+	if err != nil {
+		return err
+	}
+	var stderr bytes.Buffer
+
+	wasmCmd := exec.Command(wasmTools, "component", "wit", "--wat", "--all-features", "--output", witFilePath)
```

Comment:
This now fails if `wasm-tools` is not present.

Reply:
yeah but wouldn't generator fail before this code being executed? the `wasm-tools` will run to convert WIT to `.json`, unless the path is a `.json` file already. Perhaps we only want to generate `.json` WIT file and place it adjacent to the source code?

Reply:
Wit-bindgen-go can generate WIT from JSON. Would that work?

Reply:
Yeah see my latest change - I used Resolve to WIT to generate the wit files.

Reply:
moved this logic to #179